### DISCUSSION TITLE
How to call CoreDisplay_DisplayCreateInfoDictionary(CGDirectDisplayID) function?

### DISCUSSION THREAD

**Discussion by coolcoolkg21** [MonitorControl/MonitorControl#635] [Q&A] [answered]
Dear author,

I'm trying to build a similar project that programmed with Objective-C, but it fails and shows "Undefined symbol: _CoreDisplay_DisplayCreateInfoDictionary."
Do I have to import other header files or libraries to make this function work?
Thanks in advance.

**Comment by waydabber** [comment, marked as answer]
Hi,

yes, you should add this:

`extern CFDictionaryRef CoreDisplay_DisplayCreateInfoDictionary(CGDirectDisplayID);`

Also you probably have to link the CoreDisplay framework to your project from `/System/Library/Frameworks/CoreDisplay.framework`:

<img width="170" alt="Screen Shot 2021-10-01 at 10 16 38" src="https://user-images.githubusercontent.com/37590873/135588025-6ca812eb-cc37-4db1-9037-a8bc6dd78a21.png">

Cheers!

**Reply by coolcoolkg21** [reply]
Thank you so much!
